phpunit/2_Integration/DevicesLocksTest.php

```diff
@@ -380,4 +380,133 @@ public function switchLockItemtype() {
+      <OSVERSION>4.8.11-040811-generic</OSVERSION>
+      <PROCESSORN>1</PROCESSORN>
+      <PROCESSORS>2300</PROCESSORS>
+      <PROCESSORT>Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz</PROCESSORT>
+      <SWAP>8035</SWAP>
+      <USERID>adelaunay</USERID>
+      <UUID>4C4C4544-0042-5410-8047-B5C04F503732</UUID>
+      <VMSYSTEM>Physical</VMSYSTEM>
+      <WINPRODID>ID-1000010001</WINPRODID>
+      <WORKGROUP>ad.teclib.infra/luisant.chartres.workgroup.teclib.infra</WORKGROUP>
+    </HARDWARE>
+    <MONITORS>
+      <BASE64>AP///////wAQrFVAUzIyNQ0UAQMKLBl47u6Vo1RMmSYPUFSlSwBxT4GAqcABAQEBAQEBAQEBMCpAyGCEZDAYUBMAu/kQAAAeAAAA/wBQMTI1UjAzVTUyMlMKAAAA/ABERUxMIFAyMDEwSAogAAAA/QA4TB5TEAAKICAgICAgAOc=</BASE64>
+      <CAPTION>DELL P2010H</CAPTION>
+      <DESCRIPTION>13/2010</DESCRIPTION>
+      <MANUFACTURER>Dell Inc.</MANUFACTURER>
+      <NAME>Dell P2010H (Analog)</NAME>
+      <PORT>VGA</PORT>
+      <SERIAL>P125R03U243P</SERIAL>
+      <TYPE>Dell P2010H (Analog)</TYPE>
+    </MONITORS>
+  </CONTENT>
+  <DEVICEID>LU002-2016-05-12-10-04-59</DEVICEID>
+  <QUERY>INVENTORY</QUERY>
+</REQUEST>';
+
+      $pfCommunication->handleOCSCommunication($computer_xml, '', 'glpi');
+
+      $this->assertEquals(countElementsInTable('glpi_monitors'), 1, 'The monitor has not been created :/');
+
+      // Check logs
+      $GLPIlog->testSQLlogs();
+      $GLPIlog->testPHPlogs();
+
+      // Check computer and monitor have id 1
+      $this->assertEquals(countElementsInTable('glpi_monitors', ['id' => 1]), 1, 'The monitor has not id 1 :/');
+      $this->assertEquals(countElementsInTable('glpi_computers', ['id' => 1]), 1, 'The computer has not id 1 :/');
+
+      $monitor->update([
+         'id' => 1,
+         'name' => 'another name',
+         'serial' => 'XXXX-1-XXXX'
+      ]);
+
+      // Test replaceids and lock may not delete fields if in lock
+      $a_inventory = [
+         "Computer" => [
+            "name"   => "LU002",
+            "serial" => "5BTGP72"
+          ],
+          'monitor'        => [
+              [
+                  'name'    => 'DELL P2010H',
+                  'manufacturers_id'=> "Dell Inc.",
+                  'serial'  => 'P125R03U243P',
+                  'is_dynamic' => 1
+              ]
+          ],
+      ];
+      $manufacturers_id = $manufacturer->getFromDBByCrit(['name' => 'Dell Inc.']);
+      $a_computerinventory = $pfFormatconvert->replaceids($a_inventory, 'Computer', 1);
+      $reference = [
+         "Computer" => [
+            "name"   => "LU002",
+            "serial" => "5BTGP72"
+          ],
+          'monitor'        => [
+              [
+                  'name'    => 'DELL P2010H',
+                  'manufacturers_id'=> $manufacturer->getID(),
+                  'serial'  => 'P125R03U243P',
+                  'is_dynamic' => 1
+              ]
+          ],
+      ];
+      $this->assertEquals($a_computerinventory, $reference, 'Replaceid with lock must not delete fields');
+   }
```

Comment:
I think another test updating the computer should show the lock is working on the computer object.

Reply:
you're right, I'm currently add it

Reply:
Humm have yet it function computerManufacturerLock before this test